packages/core/src/components/Alert/Alert.jsx

```diff
@@ -1,10 +1,23 @@
 import PropTypes from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
+import uniqueId from 'lodash.uniqueid';
 
 export class Alert extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.headingId = this.props.headingId || uniqueId('alert_');
+  }
+
   heading() {
-    if (this.props.heading) {
+    if (typeof id !== 'undefined') {
```

Comment:
We're defining `this.headingId` in the constructor so we will always have the string *alert_N* available to create unique IDs. Users can override it by passing a prop `headingId` into their component when declared in JSX. I think we can return a heading without the if/else statement, maybe like this:

```javascript
heading() {
  return (
    <h3 className="ds-c-alert__heading" id={this.headingId}>
      {this.props.heading}
    </h3>
  );
}
```